### PR TITLE
[Harness] Refactor RunSimulatorTask. 

### PR DESCRIPTION
**Pull request by mandel-macaque**
Move most/all of the logic outside the task to later be able co create a
similar task in the CLI.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
As with others, draft to simplify the review process.

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8385/1/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8385/1/display/redirect) (on internal Jenkins) failed in stage(s) 'Install Provisioning Profiles' 🔥 : hudson.AbortException: script returned exit code 104

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-8385/1//console)
✅ Packages: 
* [xamarin.ios-13.21.0.50.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8385/56242c5e1af49265c2d6f3b152ed2a04a8b58afd/1/package/xamarin.ios-13.21.0.50.pkg)
* [xamarin.mac-6.21.0.50.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8385/56242c5e1af49265c2d6f3b152ed2a04a8b58afd/1/package/xamarin.mac-6.21.0.50.pkg)

File: tests/xharness/TestTasks/IRunXITask.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Xharness.TestTasks {
+	public interface IRunXITask<TDevice> : IRunTestTask where TDevice : class, IDevice {
+
+		public AppRunner Runner { get; set; }
+		public AppRunner AdditionalRunner { get; set; }
+		TestTarget AppRunnerTarget { get; set; }
+		public IEnumerable<TDevice> Candidates { get; }
+		public TDevice Device { get; set; }
+		public TDevice CompanionDevice { get; set; }
```
Comment (mandel-macaque) [inline tests/xharness/TestTasks/IRunXITask.cs:13]:
```suggestion
		AppRunner Runner { get; set; }
		AppRunner AdditionalRunner { get; set; }
		TestTarget AppRunnerTarget { get; set; }
		IEnumerable<TDevice> Candidates { get; }
		TDevice Device { get; set; }
		TDevice CompanionDevice { get; set; }
```

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8385/2/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8385/2/display/redirect) (on internal Jenkins) failed in stage(s) 'Build' 🔥 : org.jenkinsci.plugins.workflow.steps.FlowInterruptedException

🔥 [Build failed](https://jenkins.internalx.com/job/macios/job/PR-8385/2//console) 🔥

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) (on internal Jenkins) failed in stage(s) 'Running XM tests on '10.11', Running XM tests on '10.11', Running XM tests on '10.12', Running XM tests on '10.13', Running XM tests on '10.14', Running XM tests on '10.9', Running XM tests on '10.9'' 🔥

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-8385/3//console)
✅ Packages: 
* [xamarin.ios-13.21.0.52.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8385/d6273324816952f08a8554667a0ee3555ee8ee50/3/package/xamarin.ios-13.21.0.52.pkg)
* [xamarin.mac-6.21.0.52.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8385/d6273324816952f08a8554667a0ee3555ee8ee50/3/package/xamarin.mac-6.21.0.52.pkg)

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-8385/d6273324816952f08a8554667a0ee3555ee8ee50/3/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-8385/d6273324816952f08a8554667a0ee3555ee8ee50/3/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-8385/d6273324816952f08a8554667a0ee3555ee8ee50/3/jenkins-results/generator-diff/index.html) (no change)
🔥 [Xamarin.Mac tests on 10.10 failed: Xamarin.Mac tests on macOS 10.10 failed (apitest)](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) 🔥
🔥 [Xamarin.Mac tests on 10.11 failed: Xamarin.Mac tests on macOS 10.11 failed (apitest)](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) 🔥
🔥 [Xamarin.Mac tests on 10.12 failed: Xamarin.Mac tests on macOS 10.12 failed (apitest)](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) 🔥
🔥 [Xamarin.Mac tests on 10.13 failed: Xamarin.Mac tests on macOS 10.13 failed (apitest)](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) 🔥
🔥 [Xamarin.Mac tests on 10.14 failed: Xamarin.Mac tests on macOS 10.14 failed (apitest)](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) 🔥
🔥 [Xamarin.Mac tests on 10.9 failed: Xamarin.Mac tests on macOS 10.9 failed (apitest)](https://jenkins.internalx.com/job/macios/job/PR-8385/3/display/redirect) 🔥
🔥 [Test run failed](http://xamarin-storage/jenkins/xamarin-macios/PR-8385/d6273324816952f08a8554667a0ee3555ee8ee50/3/jenkins-results/tests/index.html) 🔥

# Test results

<details>
<summary>2 tests failed, 183 tests passed.</summary>

## Failed tests

 * [NUnit] Mono Mac OS X BCL tests group 2/Mac Modern/Debug: Failed (Test run failed.)
 * mmptest/macOS/Debug: Failed (Execution failed with exit code 2)
</details>

**Comment by mandel-macaque** [comment]
Failing tests are un-related and known: https://github.com/xamarin/xamarin-macios/issues/8395

**Pull request merged**